engine.py

```diff
@@ -349,6 +349,10 @@ def init_engine(self):
             self.__watcher = SceneEventWatcher(cb_fn)
             self.log_debug("Registered open and save callbacks.")
 
+        # Initialize a dictionary of Maya panels that have been created by the engine.
+        # Each panel entry has a Maya panel name key and an app widget instance value.
+        self._maya_panel_dict = {}
```

Comment:
wondering if it is worth explicitly clearing this in `destroy_engine()` just to help the GC? Can you do a quick check and see if it leaks memory on engine restart? Sometimes python/qt/pyside can be weird and wonderful with these things....